plugins/woocommerce/tests/e2e-pw/tests/merchant/order-edit.spec.js

```diff
@@ -95,7 +95,39 @@ test.describe( 'Edit order > Downloadable product permissions', () => {
 		email: 'john.doe@example.com',
 	};
 
-	let orderId, productId, product2Id, noProductOrderId;
+	let orderId,
+		productId,
+		product2Id,
+		noProductOrderId,
+		initialGrantAccessAfterPaymentSetting;
+
+	/**
+	 * Enable the "Grant access to downloadable products after payment" setting in WooCommerce > Settings > Products > Downloadable products.
+	 */
+	const enableGrantAccessAfterPaymentSetting = async ( api ) => {
+		const endpoint =
+			'settings/products/woocommerce_downloads_grant_access_after_payment';
+
+		// Get current value
+		const response = await api.get( endpoint );
+		initialGrantAccessAfterPaymentSetting = response.data.value;
+
+		// Enable
+		if ( initialGrantAccessAfterPaymentSetting !== 'yes' ) {
+			await api.put( endpoint, {
+				value: 'yes',
+			} );
+		}
+	};
+
+	const revertGrantAccessAfterPaymentSetting = async ( api ) => {
+		const endpoint =
+			'settings/products/woocommerce_downloads_grant_access_after_payment';
+
+		await api.put( endpoint, {
+			value: initialGrantAccessAfterPaymentSetting,
+		} );
+	};
```

Comment:
I added these functions because sometimes, the daily smoke test site starts with an unchecked `Grant access to downloadable products after payment` option, probably because of a failed test from previous runs. These functions ensures that this option is checked prior to starting the test, and reverts it to its previous state after the test.